biogeochem/EDCanopyStructureMod.F90

```diff
@@ -2010,15 +2007,26 @@ subroutine update_hlm_dynamics(nsites,sites,fcolumn,bc_out)
            end do
            
         endif
+
+        ! If running hydro, perform a final check to make sure that we
+        ! have conserved water. Since this is the very end of the dynamics
+        ! cycle. No water should had been added or lost to the site during dynamics.
+        ! With growth and death, we may have shuffled it around.
+        ! For recruitment, we initialized their water, but flagged them
+        ! to not be included in the site level balance yet, for they
+        ! will demand the water for their initialization on the first hydraulics time-step
+        
+        if (hlm_use_planthydro.eq.itrue) then
+           call UpdateH2OVeg(sites(s),bc_out(s),bc_out(s)%plant_stored_h2o_si,1)
+        end if
         
      end do
 
-     ! If hydraulics is turned on, update the amount of water bound in vegetation
+     ! This call is purely for diagnostics and history variables
```

Comment:
It might be good to point out the real recruit water demand is handled by subroutine RecruitWUptake(nsites,sites,bc_in,dtime,recruitflag). This is only for diagnostics and history variables. I am wondering if we should delete RecruitWaterStorage as it is redundant and it is not functional at all.

Reply:
We could potentially delete RecruitWaterStorage(), but its not straightforward.  It is called near the end of the dynamics sequence, but before the call to update history variables.  Here is the variable it informs, which is updated with the rest of the dynamics history variables:
https://github.com/NGEET/fates/blob/master/main/FatesHistoryInterfaceMod.F90#L2125

If we combined RecruitWaterStorage() into RecruitWUptake(), the update of the history variable wouldn't happen at the right time.